redis_queue.go

```diff
@@ -0,0 +1,136 @@
+package ewf
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+var _ Queue = (*RedisQueue)(nil)
+
+// RedisQueue is the Redis implementation of the Queue interface
+type RedisQueue struct {
```

Comment:
the queue should be private and constructor should return `Queue` interface because using the struct directly would leave all fields uninitialized